### PR TITLE
Separated touch and mouse delay options

### PR DESCRIPTION
**Pull request by alexandercrichton**
Hi!

Forgive me, this is my first pull request, so I'm not accustomed to the process. Also many thanks for this project as I would currently be lost without it.

My research leads me to believe that mobile detection isn't very reliable, and if possible the best solution is to handle both touch and mouse events. I've been using this backend for both touch and mouse devices. The only issue I've run into is that a single delay option isn't suitable; with the mouse you expect drags to start immeidately, but on touch devices an immediate drag interferes with scrolling.

I've just made it so separate delays can be specified for the touch and mouse drag start events. Now this backend can use a single configuration to handle both types of user controls intuitively.

### CODE REVIEW

**Comment by Yuripetusko** [comment]
Need this PR in master :)

**Comment by yahoocla** [comment]
Thank you for submitting this pull request, however I do not see a valid CLA on file for you. Before we can merge this request please visit https://yahoocla.herokuapp.com/ and agree to the terms. Thanks! :smile:

**Comment by yahoocla** [comment]
CLA is valid!

**Comment by longlho** [comment]
👍

File: src/Touch.js [hunk 1]
```diff
@@ -190,7 +196,12 @@ export class TouchBackend {
     }
 
     handleTopMoveStartDelay (e) {
-        this.timeout = setTimeout(this.handleTopMoveStart.bind(this, e), this.delay);
+        if (e.type === eventNames['touch']['start']) {
```
Comment (longlho) [inline src/Touch.js]: `eventNames.touch.start`

File: src/Touch.js [hunk 2]
```diff
@@ -55,15 +55,17 @@ export class TouchBackend {
         options = {
             enableTouchEvents: true,
             enableMouseEvents: false,
-            delay: 0,
+            delayTouchStart: 0,
```
Comment (longlho) [inline src/Touch.js]: this is not backwards-compat so you should support `delay` still

**Comment by longlho** [comment]
@alexandercrichton left a couple of comments

**Comment by alexandercrichton** [comment]
Thanks @longlho! Still learning this PR business. I gather my last commit is now part of this PR?

I've tidied up your first comment and retained support for the original delay option

**Comment by longlho** [comment]
@alexandercrichton looks good :) 1 last step is to squash the 2 commits (`git rebase -i HEAD~2`) then I'll merge

**Comment by alexandercrichton** [comment]
Ok I think I've done that right

**Comment by longlho** [comment]
👍

**Pull request closed without merging**